tests/test_server/test_grpc/test_training_servicer.py

```diff
@@ -0,0 +1,498 @@
+from tiktorch.server.grpc import training_servicer
+from tiktorch.server.session.backend.base import TrainerSessionBackend
+from tiktorch.server.session.process import TrainerSessionProcess
+from tiktorch.server.session_manager import SessionManager
+from tiktorch.trainer import ShouldStopCallbacks, Trainer, TrainerState
+
+
+@pytest.fixture(scope="module")
+def grpc_add_to_server():
+    return training_pb2_grpc.add_TrainingServicer_to_server
+
+
+@pytest.fixture(scope="module")
+def grpc_servicer():
+    return training_servicer.TrainingServicer(TorchDevicePool(), SessionManager())
+
+
+@pytest.fixture(autouse=True)
+def clean(grpc_servicer):
+    yield
+    grpc_servicer.close_all_sessions()
+
+
+@pytest.fixture(scope="module")
+def grpc_stub_cls():
+    return training_pb2_grpc.TrainingStub
+
+
+def unet2d_config_path(checkpoint_dir, train_data_dir, val_data_path, device: str = "cpu"):
+    return f"""
+device: {device}  # Use CPU for faster test execution, change to 'cuda' if GPU is available and necessary
+model:
+  name: UNet2D
+  in_channels: 3
+  out_channels: 2
+  layer_order: gcr
+  f_maps: 16
+  num_groups: 4
+  final_sigmoid: false
+  is_segmentation: true
+trainer:
+  checkpoint_dir: {checkpoint_dir}
+  resume: null
+  validate_after_iters: 2
+  log_after_iters: 2
+  max_num_epochs: 1000
+  max_num_iterations: 10000
+  eval_score_higher_is_better: True
+optimizer:
+  learning_rate: 0.0002
+  weight_decay: 0.00001
+loss:
+  name: CrossEntropyLoss
+eval_metric:
+  name: MeanIoU
+  ignore_index: null
+lr_scheduler:
+  name: MultiStepLR
+  milestones: [2, 3]
+  gamma: 0.5
+loaders:
+  dataset: StandardHDF5Dataset
+  batch_size: 1
+  num_workers: 1
+  raw_internal_path: raw
+  label_internal_path: label
+  weight_internal_path: null
+  train:
+    file_paths:
+      - {train_data_dir}
+
+    slice_builder:
+      name: SliceBuilder
+      patch_shape: [1, 64, 64]
+      stride_shape: [1, 64, 64]
+      skip_shape_check: true
+
+    transformer:
+      raw:
+        - name: Standardize
+        - name: RandomFlip
+        - name: RandomRotate90
+        - name: RandomRotate
+          axes: [[2, 1]]
+          angle_spectrum: 30
+          mode: reflect
+        - name: ElasticDeformation
+          execution_probability: 1.0
+          spline_order: 3
+        - name: AdditiveGaussianNoise
+          execution_probability: 1.0
+        - name: AdditivePoissonNoise
+          execution_probability: 1.0
+        - name: ToTensor
+          expand_dims: true
+      label:
+        - name: RandomFlip
+        - name: RandomRotate90
+        - name: RandomRotate
+          axes: [[2, 1]]
+          angle_spectrum: 30
+          mode: reflect
+        - name: ElasticDeformation
+          execution_probability: 1.0
+          spline_order: 0
+        - name: ToTensor
+          # do not expand dims for cross-entropy loss
+          expand_dims: false
+          # cross-entropy loss requires target to be of type 'long'
+          dtype: 'long'
```

Comment:
if you change the `'long'` here and further down, to `'int64'` the windows tests will likely pass.

```suggestion
          dtype: 'int64'
```

Background:
> Some types, such as [numpy.int_](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.int_) and [numpy.intp](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.intp), have differing bitsizes, dependent on the platforms (e.g. 32-bit vs. 64-bit CPU architectures). This should be taken into account when interfacing with low-level code (such as C or Fortran) where the raw memory is addressed.

(from [numpy docs](https://numpy.org/doc/stable/user/basics.types.html#numerical-data-types))

`'long'` is an alias for `numpy.int_` and on windows and will result in 32 bit integers - which breaks the cross-entropy calculation, that expects int64...

Reply:
Yep, it worked! Awesome!